packages/flame/lib/src/effects/effect.dart

```diff
@@ -25,7 +25,7 @@ import 'package:meta/meta.dart';
 /// changes in the effect's target; and also the `reset()` method if they have
 /// non-trivial internal state.
 abstract class Effect extends Component {
-  Effect(this.controller)
+  Effect(this.controller, {this.onFinishCallback})
```

Comment:
I don't remember what our reason was for calling this `onFinishCallback`, when similar functions are called `onComplete` in the rest of the code base. It had something to do with the previous effects system I think, do you remember @st-pasha? Maybe this is a good time to rename this to `onComplete` and deprecate `onFinishCallback`.

Reply:
I am onboard if we deprecate before fully removing!

Reply:
I don't remember why it was called that way, `onComplete` seems perfectly fine to me.